lib/knife-solo/berkshelf.rb

```diff
@@ -19,7 +19,12 @@ def install!
       path = berkshelf_path
       ui.msg "Installing Berkshelf cookbooks to '#{path}'..."
 
-      berksfile = ::Berkshelf::Berksfile.from_file('Berksfile')
+      if defined?(::Berkshelf) && Gem::Version.new(::Berkshelf::VERSION) >= Gem::Version.new("3.0.0")
```

Comment:
Didn't realize you'd added a compatibility check. Thanks!

What's the reason for the `defined?(::Berkshelf)` check here? Seems like if it failed the original code as well as line 26 here will fail as well.